code/game/machinery/doors/firedoor.dm

```diff
@@ -135,10 +136,35 @@
 		return
 
 	if(density)
+		being_held_open = TRUE
 		open()
+		handle_held_open_adjacency(user)
 	else
 		close()
 
+/obj/machinery/door/firedoor/try_to_crowbar_secondary(obj/item/I, mob/user)
+	if(welded || operating)
+		return
+
+	if(density)
+		open()
+	else
+		close()
+
+/obj/machinery/door/firedoor/proc/handle_held_open_adjacency(mob/user)
+	var/mob/living/living_user = isliving(user) ? user : null
```

Comment:
If you're returning when `!isliving(user)` anyway, then you don't need this at all.

```suggestion
	var/mob/living/living_user = user
```